tidy3d/components/medium.py

```diff
@@ -82,54 +82,218 @@ def _eps_model(self, frequency: float) -> complex:
 """ Medium Definitions """
 
 
-class NonlinearSpec(ABC, Tidy3dBaseModel):
-    """Abstract specification for adding a nonlinearity to a medium.
+class NonlinearModel(ABC, Tidy3dBaseModel):
+    """Abstract model for a nonlinear material response.
+    Used as part of a :class:`.NonlinearSpec`."""
+
+    def _validate_medium_type(self, medium: AbstractMedium):
+        """Check that the model is compatible with the medium."""
+        if isinstance(medium, AbstractCustomMedium):
```

Comment:
also not support time modulated medium: `if isinstance(medium, AbstractCustomMedium) or medium.time_modulated`?